bd-logger/src/async_log_buffer.rs

```diff
@@ -984,6 +1012,57 @@ impl<R: LogReplay + Send + 'static> AsyncLogBuffer<R> {
     }
   }
 
+  fn start_file_watcher_if_needed(&mut self, crash_monitor: &Monitor) {
+    self.crash_monitor = Some(crash_monitor.clone());
+
+    if self.current_session_file_rx.is_none() {
+      let (file_tx, file_rx) = mpsc::channel(REPORT_FILES_WATCHER_CHANNEL_BUFFER_SIZE);
+      self.current_session_file_rx = Some(file_rx);
+
+      match crash_monitor.start_file_watcher(file_tx) {
+        Ok(watcher) => {
+          self.file_watcher = Some(watcher);
+          log::debug!("Started file watcher for current session reports");
+        },
+        Err(e) => {
+          log::warn!("Failed to start file watcher: {e}");
+        },
+      }
+    }
+  }
+
+  async fn process_watcher_file(&mut self, file_path: std::path::PathBuf) {
+    let Some(ref monitor) = self.crash_monitor else {
+      log::warn!(
+        "File watcher detected file but monitor not available: {}",
+        file_path.display()
+      );
+      return;
+    };
```

Comment:
This case would perma drop the report with the new setup, ideally we'd make this case impossible